### ISSUE TITLE
rethrowing error from worker causes illegal instruction panic

### ISSUE THREAD

**Issue by trvswgnr** [oven-sh/bun#15415]
### How can we reproduce the crash?

1. `git clone https://github.com/trvswgnr/bun-worker-microservices.git`
2. `git checkout bun-crash-bug`
3. bun ./index.ts
4. see intentional error thrown at `services/logger.ts:7:9`
5. see error being rethrown at `shared/util.ts:102:13`

### Relevant log output

```shell
➜ worker-microservices git:(main) ✗ bun ./index.ts
123 travvy
2 | import fs from "node:fs";
3 | 
4 | const validStrategies = {
5 |     console: console.log,
6 |     file: (msg: string) => {
7 |         fs.appendFileSync("logs/app.log", `${msg}\n`)
               ^
ENOENT: No such file or directory
   errno: -2
 syscall: "open"

      at file (/Users/WagnerT/projects/worker-microservices/services/logger.ts:7:12)
      at /Users/WagnerT/projects/worker-microservices/services/logger.ts:23:41
      at log (/Users/WagnerT/projects/worker-microservices/services/logger.ts:29:18)
      at /Users/WagnerT/projects/worker-microservices/shared/util.ts:88:37
============================================================
Bun v1.1.36 (ededc168) macOS x64
macOS v13.2.1
CPU: sse42 popcnt avx avx2
Args: "bun" "./index.ts"
Features: jsc transpiler_cache(16) tsconfig workers_spawned(15) 
Builtins: "bun:main" "node:fs" "node:string_decoder" "node:util/types" 
Elapsed: 80ms | User: 413ms | Sys: 144ms
RSS: 99.09MB | Peak: 99.09MB | Commit: 1.07GB | Faults: 146

panic: Illegal instruction at address 0x10220BF53
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.36/ma1ededc16EgggkggD__m6i/iBo4orlBirgglBy3+/kBmo+/kB+usmmBipisagy3xIA3Cm1/ikC

234 travvy
2 | import fs from "node:fs";
3 | 
4 | const validStrategies = {
5 |     console: console.log,
6 |     file: (msg: string) => {
7 |         fs.appendFileSync("logs/app.log", `${msg}\n`)
               ^
ENOENT: No such file or directory
   errno: -2
 syscall: "open"

      at file (/Users/WagnerT/projects/worker-microservices/services/logger.ts:7:12)
      at /Users/WagnerT/projects/worker-microservices/services/logger.ts:23:41
      at log (/Users/WagnerT/projects/worker-microservices/services/logger.ts:29:18)
      at /Users/WagnerT/projects/worker-microservices/shared/util.ts:88:37
[1]    54576 illegal hardware instruction  bun ./index.ts
```

### Stack Trace (bun.report)

Bun v1.1.36 ([`ededc16`](<https://github.com/oven-sh/bun/tree/ededc168cf07afcdbe5ca34495c58df435ea526d>)) on macos x86_64 [AutoCommand]

Illegal instruction at address 0x106A58F53

- *2 unknown/js code*
- `WebCore::reportException`
- `WebCore::JSEventListener::handleEvent`
- `WebCore::EventTarget::innerInvokeEventListeners`
- `WebCore::EventTarget::fireEventListeners`
- `WebCore::EventTarget::dispatchEvent`
- `WebWorker__dispatchError`
- `src.bun.js.web_worker.WebWorker.onUnhandledRejection`
- `Bun__reportUnhandledError`

Features: workers\_spawned, jsc, transpiler\_cache, tsconfig

<sub>Sentry Issue: <strong><a href="https://bun-p9.sentry.io/issues/6090825808/">BUN-8Y3</a></strong></sub>

**Comment by trvswgnr** [comment]
yes

**Comment by Jarred-Sumner** [comment]
Tracking this in https://github.com/oven-sh/bun/issues/15964

TLDR: `Worker` in Bun is still unstable, especially when it comes to error handling or closing them.
